frontend/src/components/core/StreamlitDialog/styled-components.ts

```diff
@@ -67,12 +67,16 @@ export const StyledFullRow = styled.div(({ theme }) => ({
   gap: theme.spacing.sm,
 }))
 
-export const StyledHeader = styled.h4(({ theme }) => ({
+export const StyledHeader = styled.h2(({ theme }) => ({
   paddingBottom: 0,
   paddingTop: 0,
   marginBottom: 0,
-  marginTop: 0,
-  lineHeight: 1.25,
+  marginTop: "0 !important",
```

Comment:
I think the `!important` flag feels like a band-aid for future problem. I'd rather override the proper style and highlight where the problem is coming from.

Really it's the global styles it's the `:first-of-type`, so if you add:

```
  ...
  marginTop: 0,
  ...
  // Override the default global style for a h2:first-of-type
  "&:first-of-type": {
    marginTop: 0,
  },

```

I keep the original set to 0, for all other h2s that might apply, but I mark out that the override is from the globalStyles.

Probably the real solution is to scope the globalStyles to be simpler. This style seems more required for our Markdown component.